packages/gatsby/src/query/index.js

```diff
@@ -38,54 +36,56 @@ const popExtractedQueries = () => {
 }
 
 const findIdsWithoutDataDependencies = state => {
-  const allTrackedIds = _.uniq(
-    _.flatten(
-      _.concat(
-        _.values(state.componentDataDependencies.nodes),
-        _.values(state.componentDataDependencies.connections)
-      )
-    )
+  const allTrackedIds = new Set(
+    _.flatten([
+      ...state.componentDataDependencies.nodes.values(),
+      ...state.componentDataDependencies.connections.values(),
+    ])
```

Comment:
This seems to be problematic:
this PR on the left vs latest/master on the right
![Screenshot 2019-09-20 at 18 59 49](https://user-images.githubusercontent.com/419821/65344827-092f6d00-dbd9-11e9-8b62-740ab4fef2f7.png)

Currently it doesn't properly flatten our data structure, which means that it deems all ids to be untracked and cause to run all queries

Reply:
Alternative to change above is:
```js
  const allTrackedIds = new Set()
  const boundAddToTrackedIds = allTrackedIds.add.bind(allTrackedIds)
  state.componentDataDependencies.nodes.forEach(dependenciesOnNode => {
    dependenciesOnNode.forEach(boundAddToTrackedIds)
  })
  state.componentDataDependencies.connections.forEach(
    dependenciesOnConnection => {
      dependenciesOnConnection.forEach(boundAddToTrackedIds)
    }
  )
```
Which seems to the trick (at least for 1to1 conversion with current algorithm)

Reply:
Done